vault/data_source_generic_secret_list_v2.go

```diff
@@ -0,0 +1,28 @@
+package vault
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func genericSecretListDataSourceV2() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: genericSecretListDataSourceRead,
+
+		Schema: map[string]*schema.Schema{
+			"path": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Full KV-V2 path where secrets will be listed.",
+				ValidateFunc: validateNoTrailingSlash,
+			},
+
+			"names": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: "List of all secret names.",
```

Comment:
nit: include the relationship of `names` to `path`, you may also want to consider sharing the schema between the two data sources.
```suggestion
				Description: "List of all secret names found under  path",
```